ansible/roles/test/files/ptftests/vlan_test.py

```diff
@@ -0,0 +1,268 @@
+    #--------------------------------------------------------------------------
+
+    def setUp(self):
+        self.vlan_ports_list = ast.literal_eval(self.test_params["vlan_ports_list"])
+        self.vlan_intf_list = ast.literal_eval(self.test_params["vlan_intf_list"])
+        self.router_mac = self.test_params["router_mac"]
+
+        for vlan_port in self.vlan_ports_list:
+            vlan_port["pvid"] = int(vlan_port["pvid"])
+            vlan_port["port_index"] = int(vlan_port["port_index"])
+
+        self.dataplane = ptf.dataplane_instance
+        self.test_params = test_params_get()
+        self.setUpArpResponder()
+        self.log("Start arp_responder")
+        self.shell(["supervisorctl", "start", "arp_responder"])
+
+        self.log("Create VLAN intf")
+        for vlan_port in self.vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) != vlan_port["pvid"]:
+                    self.shell(["ip", "link", "add", "link", "eth%d"%vlan_port["port_index"], 
+                                "name", "eth%d.%s"%(vlan_port["port_index"], permit_vlanid), 
+                                "type", "vlan", "id", str(permit_vlanid)])
+                    self.shell(["ip", "link", "set", 
+                                "eth%d.%s"%(vlan_port["port_index"], permit_vlanid), "up"])
+
+        logging.info("VLAN test starting ...")
+        pass
+    #--------------------------------------------------------------------------
+
+    def setUpArpResponder(self):
+        vlan_ports_list = self.vlan_ports_list
+        d = defaultdict(list)
+        for vlan_port in vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) == vlan_port["pvid"]:
+                    iface = "eth%d" % vlan_port["port_index"]
+                else:
+                    iface = "eth%d.%s" % (vlan_port["port_index"], permit_vlanid)
+                d[iface].append(vlan_port["permit_vlanid"][str(permit_vlanid)]["peer_ip"])
+        with open('/tmp/from_t1.json', 'w') as file:
+            json.dump(d, file)
+
+    #--------------------------------------------------------------------------
+    def tearDown(self):
+        logging.info("VLAN test ending ...")
+
+        self.log("Delete VLAN intf")
+        for vlan_port in self.vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) != vlan_port["pvid"]:
+                    self.shell(["ip", "link", "delete", "eth%d.%d"%(vlan_port["port_index"], int(permit_vlanid))])
+
+        self.log("Stop arp_responder")
+        self.shell(["supervisorctl", "stop", "arp_responder"])
+
+        pass
+
+    #--------------------------------------------------------------------------
+    def build_icmp_packet(self, vlan_id, 
+                          src_mac="00:22:00:00:00:02", dst_mac="00:22:00:00:00:01", 
```

Comment:
where does this dst_mac get populated the in the DUT switch. How does the switch know how to forward this packet?

Reply:
If dst_mac is not specified, an unknown unicast ICMP package is constructed by default. "00:22:00:00:00:01" is unknown for switch unless a coincidence, switch need to flood this packet in the VLAN.

Reply:
since your purpose is to broadcast the packet, why not use broadcast mac in this case? the behavior of unknown unicast mac can be changed.